File: pinyougou-web/pinyougou-shop-web/src/main/java/com/pinyougou/shop/controller/IndexController.java
package com.pinyougou.shop.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.service.SellerService;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/index")
public class IndexController {
    @GetMapping("/showUsername")
    public Map<String,String> showUsername() {
        try {
            //获取用户名
            String username = SecurityContextHolder.getContext().getAuthentication().getName();
            Map<String, String> data = new HashMap<>();
            data.put("username",username);
            return data;
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return null;
    }
}

File: pinyougou-web/pinyougou-shop-web/src/main/webapp/js/controller/indexController.js
/** 定义控制器层 */
app.controller('indexController', function($scope, baseService){
    $scope.showUsername = function () {
        //发送异步请求
        baseService.sendGet('/index/showUsername').then(function (response) {
            $scope.username = response.data.username;
        })
    }


});
